subsys/net/ip/tcp2.c

```diff
@@ -1275,6 +1279,110 @@ static enum net_verdict tcp_recv(struct net_conn *net_conn,
 	return NET_DROP;
 }
 
+static uint32_t seq_scale(uint32_t *seq)
+{
+	return *seq + (k_ticks_to_ns_floor32(k_uptime_ticks()) >> 6);
+}
+
+static uint8_t secret_key[16]; /* MD5 128 bits as described in RFC6528 */
```

Comment:
I see that "secret key" terminology comes from RFC6528, but it's clear that there's only so much secrecy in it. It's for example not comparable to TLS secret keys. I'd suggest to consider calling it `unique_key`, or if there's desire to follow the RFC6528 terminology, to `rfc6528_secretkey`, to give clear context of its "secretness".